### PR TITLE
Replace window resize listener with more holistic solution

### PR DESCRIPTION
**Pull request by jnatalzia**
### This PR will...
Replace the current responsive resize listener with one that adapts to player container changes

### Why is this Pull Request needed?
We do not track resizes of the player if it is `100%` width and a parent container changes size

### Are there any points in the code the reviewer needs to double check?
Yes. I double checked the listener was only firing once and checked window resize and parent conditions but could be some perf impact I'm overlooking

### Are there any Pull Requests open in other repos which need to be merged with this?
Nope

#### Addresses Issue(s):

JW8-2383

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  🗿 Set a milestone. It should be the ticket's fix version in JIRA.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by jnatalzia** [comment]
test this please

File: src/js/view/utils/views-manager.js [hunk 1]
```diff
@@ -103,7 +102,6 @@ if (isAndroidChrome && hasOrientation) {
 window.addEventListener('beforeunload', () => {
     document.removeEventListener('visibilitychange', onVisibilityChange);
     document.removeEventListener('webkitvisibilitychange', onVisibilityChange);
-    window.removeEventListener('resize', scheduleResponsiveRedraw);
     window.removeEventListener('orientationchange', scheduleResponsiveRedraw);
```
Comment (jnatalzia) [inline src/js/view/utils/views-manager.js]: left this in - but debating it. not sure if it's redundant like `resize` with the changes here. 

also are there concerns with moving from the `scheduleResponsiveRedraw` function here to `_responsiveListener` (which also schedules in a way)

Comment (robwalch) [reply src/js/view/utils/views-manager.js]: `scheduleResponsiveRedraw` notifies all players that they need to check if they've been resized. If each player has it's own `ResizeListener` then `scheduleResponsiveRedraw` and the "orientationchange" listener are no longer needed.

Comment (jnatalzia) [reply src/js/view/utils/views-manager.js]: Will remove!

File: src/js/view/utils/resize-listener.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { requestAnimationFrame, cancelAnimationFrame } from 'os/utils/request-animation-frame';
-import { createElement } from 'os/utils/dom';
-import { css, style } from 'os/utils/css';
+import { requestAnimationFrame, cancelAnimationFrame } from 'utils/request-animation-frame';
+import { createElement } from 'utils/dom';
+import { css, style } from 'utils/css';
```
Comment (robwalch) [inline src/js/view/utils/resize-listener.js]: good catch

File: src/js/view/view.js [hunk 3]
```diff
@@ -297,6 +299,10 @@ function View(_api, _model) {
             viewsManager.observe(_playerElement);
         }
         _model.set('inDom', inDOM);
+
+        if (!this.resizeListener) {
```
[the remark this answers is not in the document]
Comment (jnatalzia) [reply src/js/view/view.js]: could the latter suggestion backfire if the user calls `resize` with a `%` based width? is that a possible use of the api?

[the remark this answers is not in the document]
Comment (robwalch) [reply src/js/view/view.js]: Well at that point you would need to instantiate a ResizeListener.

**Comment by jnatalzia** [comment]
@robwalch did some light memory profiling on the weight of an unused resize listener.

If I'm reading this right (and I def could not be, these tools can be tough to read) it's a little over a kb in memory to house the listener and all references associated with it. 

Screen Shot 2019-05-09 at 9 24 22 AM

Given we're removing two window listeners and a function in memory I'm of the mind we're basically replacing the memory we took out. Not as great as removing it entirely for cases where it's not needed however I am still concerned about the implications of tearing out a window resize listener that always existed and not having something to replace it in the static width case. I also think with the current flex behavior of the player (as well as potentially behavior we don't know about) it's far more risky and not work the 1kb to try and find all the cases where a resize listener wouldn't be required.

Happy to circle back around, I want to make sure this is in a good place even if it means sorting through those edge cases but figured I'd show my findings first and get your opinion

**Comment by robwalch** [comment]
@jnatalzia It a good trade off. The thing to keep in mind is that the window and orientation listeners were attached to the singleton views-manager and serviced all players. If someone creates 10 players, with the new resize listener each player will get one. I think that's OK since each will be fully responsive with the new element listener.

**Comment by robwalch** [comment]
@jnatalzia Here's a good page to use to analyze responsive performance with multiple players 

http://player-develop-test-jenkins.longtailvideo.com/builds/lastSuccessfulBuild/archive/test/public/moviepilot/

PR build
http://player-pr-test-jenkins.longtailvideo.com/builds/21485/archive/test/public/moviepilot/

File: src/js/view/utils/views-manager.js [hunk 4]
```diff
@@ -37,27 +35,6 @@ function matchIntersection(entry, group) {
     }
 }
 
-function scheduleResponsiveRedraw() {
```
Comment (robwalch) [inline src/js/view/utils/views-manager.js]: Could we do something where the ResizeListener's callbacks were debounced calls to this method?

**Comment by jnatalzia** [comment]
test this please

**Pull request merged**